src/scheduler.rs

```diff
@@ -78,11 +92,28 @@ async fn scheduler_loop(
     checker: Arc<impl Checker + 'static>,
     producer: Arc<impl ResultsProducer + 'static>,
     shutdown: CancellationToken,
+    progress_key: String,
+    redis_cluster_nodes: Vec<String>,
 ) {
-    let mut interval = time::interval(time::Duration::from_secs(1));
-
-    let start = Utc::now();
-    let instant = Instant::now();
+    let client = ClusterClient::new(redis_cluster_nodes).unwrap();
+    let mut connection = client
+        .get_async_connection()
+        .await
+        .expect("Unable to connect to Redis");
+    let result: Option<String> = connection
+        .get(&progress_key)
+        .await
+        .expect("Unable to get last tick key");
+    let tick_frequency = Duration::from_secs(1);
+    let mut current_tick_time = match result {
+        Some(result) => {
+            let nanos: i64 = result
+                .parse()
+                .expect("Last tick timestamp has invalid format");
```

Comment:
This will crash the scheduler? Do we want that or should we just return `Utc::now`